File: src/main/java/model/Message.java
package model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.annotations.Cascade;
import org.hibernate.envers.Audited;
import org.hibernate.envers.RelationTargetAuditMode;

import javax.persistence.*;

@Entity
@Table(name = "messages")
@Data
@NoArgsConstructor(force = true)
@RequiredArgsConstructor
@Audited
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "messageText", nullable = false)
    @NonNull
    private String messageText;

    @ToString.Exclude
    @JsonIgnore
    @ManyToOne
    @NonNull
    @Audited(targetAuditMode = RelationTargetAuditMode.NOT_AUDITED)
    private User user;

}

File: src/main/webapp/scripts/model/message.js
class Message {
    constructor(id, messageText, user) {
        this.id = id;
        this.messageText = messageText;
        this.user = user;
    }

    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get messageText() {
        return this._messageText;
    }

    set messageText(value) {
        this._messageText = value;
    }

    get user() {
        return this._user;
    }

    set user(value) {
        this._user = value;
    }

    getUserId() {
        return this.user.id;
    }
}

export {Message};
